### ISSUE TITLE
Next.js - only 10 pages are pre-rendered due to default limit of 10 results in GraphQL search query

### ISSUE THREAD

**Issue by jln-dk** [Sitecore/jss#630]
## Description
When running JSS+Next.js in production, the purpose of the [`sitemapFetcher`](https://github.com/Sitecore/jss/blob/master/samples/nextjs/src/lib/sitemap-fetcher.ts) in [`getStaticPaths`](https://github.com/Sitecore/jss/blob/9d1fbbfadd614ff115dcf5d8209b009f7cde9fc1/samples/nextjs/src/pages/%5B%5B...path%5D%5D.tsx#L58) in `src/pages/[[...path]].tsx` is to gather the entire sitemap and pre-render all the pages of the site. 
But that is not happening. Only 10 pages are being pre-rendered.

When executing the `search()` query on the GraphQL service it only returns 10 results. In order to increase the number of results, you must explicitly set the `first` parameter to a higher number. 

The ["search" query in `graphql-sitemap-service`](https://github.com/Sitecore/jss/blob/9d1fbbfadd614ff115dcf5d8209b009f7cde9fc1/packages/sitecore-jss-nextjs/src/services/graphql-sitemap-service.ts#L228) does not set the "first" parameter and therefore only 10 results/pages are pre-rendered.

## Expected behavior
When running in production I am expecting all pages to be pre-rendered and not only the first 10 results from the search query.

## Steps To Reproduce
1. Create a site with 10+ pages
2. Run Next.js in production
3. Look at the static build output. Only 10 pages will be pre-rendered.

## Possible Fix
Increase the default "first" limit, explicitly set it to a higher number in the default implementation of `graphql-sitemap-service`, or remove the limit totally.

## Your Environment
* Sitecore Version: 10
* JSS Version: 16

**Comment by ambrauer** [comment]
Hi @jln-dk - 

This is a known issue and will be resolved in the next release. The current cursor-based pagination will remain on the GraphQL Edge schema (likely with 10 as default); however, the JSS SDK `GraphQLSitemapService` will properly paginate/aggregate results (and possibly bump to a larger default page size). 

In the meantime, you should be able to override the query and specify a larger `first` value. For example, in the sample `sitemap-fetcher.ts`:

```javascript
return this._graphqlSitemapService.fetchSSGSitemap(
  context?.locales || [],
  this.GRAPHQL_ROOT_ITEM_PATH,
  (rootItemId: string, locale: string) => {
	return `search(
	  first: 100,
	  where: {
		AND:[
		  {
			name:"_path",
			value:"${rootItemId.toLowerCase()}"
		  },
		  {
			name:"_language",
			value:"${locale}"
		  },
		  {
			name:"_hasLayout",
			value :"true"
		  }
		]
	  }
	)`;
  }
);
```

**Comment by jln-dk** [comment]
Thanks a lot, @ambrauer. We will use your proposed fix until the next release 👍

**Comment by ambrauer** [comment]
Resolved with 19.0.0 release.
